donate/payments/tasks.py

```diff
@@ -245,6 +252,82 @@ def process_charge_succeeded(self, event):
             transaction_fee,
         )
 
+        if settings.MIGRATE_STRIPE_SUBSCRIPTIONS_ENABLED and 'thunderbird' not in metadata:
+            MigrateStripeSubscription().process(charge, subscription)
+
+
+class MigrateStripeSubscription:
+    def process(self, charge, subscription):
+        last4 = subscription.customer.sources.data[0].last4
+
+        # Fetch the Customer object that was migrated to Braintree from Stripe
+        # along with associated payment information
+        BT_customers = gateway.customer.search(
+            braintree.TransactionSearch.customer_email == charge.customer_email,
+            braintree.TransactionSearch.credit_card_number.ends_with(last4)
+        )
+
+        # Ensure we found a customer record, if not, report it to Sentry as an error
+        if len(BT_customers) == 0:
+            logger.error(f'No Customer record exists in Braintree '
+                         f'for the Stripe customer attached to this charge object:'
+                         f'{charge.id}', exc_info=True)
+            return
+
+        # There should be a single Customer/Card combo on Braintree
```

Comment:
Should we also add an `if(len(BT_customers) > 1):` here, with a sentry warning?

Reply:
Added in a check and report for this.